.github/workflows/e2e-test.yml

```diff
@@ -0,0 +1,50 @@
+# Integration and end to end tests which runs locally and deploys the Operator to a Kubernetes
+# (Minikube) cluster and creates custom resources to verify the operator's functionality
+name: Integration & End to End tests
+on:
+  pull_request:
+    branches: [ main ]
+  push:
+    branches:
+      - main
+
+jobs:
+  sample_operators_tests:
+    strategy:
+      matrix:
+        sample_dir:
+          - "sample-operators/mysql-schema"
+          - "sample-operators/tomcat-operator"
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Setup Minikube-Kubernetes
+        uses: manusa/actions-setup-minikube@v2.4.3
+        with:
+          minikube version: v1.24.0
+          kubernetes version: v1.23.0
+          github token: ${{ secrets.GITHUB_TOKEN }}
+          driver: docker
+
+      - name: Set up Java and Maven
+        uses: actions/setup-java@v2
+        with:
+          java-version: 11
+          distribution: adopt-hotspot
```

Comment:
Is using a different distribution from the tests on purpose here? Might actually be a good idea since it improves somewhat VM coverage?

Reply:
`adopt` rebranded the distribution to `temurin`, so adopt v17 doesn't exists (it's `temurin` now).

Switched to `temurin`, I "think" that, in this repository, we are not leveraging sensible parts of the JVM and we should not need to test against different distributions ... but open to suggestions!